lib/entities/workflow.ts

```diff
@@ -0,0 +1,132 @@
+import { freezeSys, toPlainObject } from 'contentful-sdk-core'
+import copy from 'fast-copy'
+import {
+  BasicMetaSysProps,
+  DefaultElements,
+  GetSpaceEnvironmentParams,
+  GetWorkflowParams,
+  Link,
+  MakeRequest,
+  PaginationQueryOptions,
+  SysLink,
+} from '../common-types'
+import { wrapCollection } from '../common-utils'
+import enhanceWithMethods from '../enhance-with-methods'
+
+export type WorkflowSysProps = Pick<
+  BasicMetaSysProps,
+  'id' | 'version' | 'createdAt' | 'createdBy' | 'updatedAt' | 'updatedBy'
+> & {
+  type: 'Workflow'
+  space: SysLink
```

Comment:
I'm wondering when do we use `SysLink` and when do we use an explicit type with `Link<T>`? I see that we do this in other places as well. Is there some pattern to it?

Reply:
Sys link doesn't enforce the type, as the Link does, in cases where it's less obvious by the key (like space) I think it makes more sense to be more restrictive and allow only a specific type.